### PR TITLE
Measure 5xx errors on all requests for image-builder-composer/worker

### PR DESCRIPTION
**Pull request by kingsleyzissou**
This pull request includes:

Creates a custom middleware function to record all 5xx errors for composer & worker as we
are currently only recording internal errors for the `/compose` endpoint.

The existing compose metrics can be removed in a follow up PR once the relevant alerts and dashboards
have been implemented.

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

File: internal/prometheus/middleware.go [hunk 1]
```diff
@@ -18,3 +20,35 @@ func MetricsMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
...
+			err := next(ctx)
+
+			path := pathLabel(ctx.Path())
+			method := ctx.Request().Method
+			status := ctx.Response().Status
+
+			httpErr := new(echo.HTTPError)
+			if errors.As(err, &httpErr) {
+				status = httpErr.Code
+			}
```
Comment (croissanne) [inline internal/prometheus/middleware.go:42]: Does the code in the echo error sometimes differ from the status one?

Also any reason for using As over Is?

Comment (kingsleyzissou) [reply internal/prometheus/middleware.go:42]: So the status code before that check is always `200`. If I used Is over As, it doesn't pick it up and the error code stays `200`.

**Comment by kingsleyzissou** [comment]
I think I found a bug with this. Marking as draft while I look into it

**Review by thozza — APPROVED** [review]
All looks good, nice work.

**Pull request merged**